### ISSUE TITLE
Windows CI: ./ci/run_envoy_docker.sh should allow users to override Docker options

### ISSUE THREAD

**Issue by sunjayBhatia** [envoyproxy/envoy#13272]
In https://github.com/envoyproxy/envoy/pull/13222 we collapse the original Windows specific script to run an Envoy build in Docker into the Linux one. We found that the CI pipelines pass a Linux specific value for `ENVOY_DOCKER_OPTIONS` which of course are invalid for Windows.

We would like to be able to use the same mechanism `ENVOY_DOCKER_OPTIONS` for overriding options in Docker on Windows (for a uniform developer experience to allow CI/local builds to add options, particularly `--storage-opt` to allow for much larger disk space).

Currently `ENVOY_DOCKER_OPTIONS` is ignored on Windows to accommodate CI, but if we can change the variable groups in CI to not pass the Linux specific options to Windows, we can stop ignoring the variable.

**Comment by wrowe** [comment]
This ask is not specific to area/windows (Sunjay you might want to modify the title to reflect this.)

**Comment by sunjayBhatia** [comment]
I think `ENVOY_DOCKER_OPTIONS` is already usable on linux unless something has changed?

**Comment by wrowe** [comment]
It is, you are asking that it can be overridden by any developer (or ci pipeline) in any environment, no?

**Comment by sunjayBhatia** [comment]
right, this issue is an ask to see if we can get the Azure Pipelines setup to not pass linux specific options on the Windows jobs for the `ENVOY_DOCKER_OPTIONS` variable in the pipeline
